bundles/admin/admin-layereditor/view/LayerWizard.jsx

```diff
@@ -1,100 +1,106 @@
-                <Steps current={this.getStep()}>
-                    <Step title={typeTitle} />
-                    <Step title="Service" />
-                    <Step title="Layers" />
-                    <Step title="Details" />
-                </Steps>
-                { this.isStep(0) &&
-                    <LayerTypeSelection
-                        types={service.getLayerTypes()}
-                        onSelect={(type) => mutator.setType(type)} />
-                }
-                { this.isStep(1) &&
-                    <div>
-                        <LayerURLForm
-                            layer={layer}
-                            loading={service.isLoading()}
-                            service={mutator} />
-                        <hr/>
-                        <Button onClick={() => this.setStep(0)}>Back</Button>
-                    </div>
-                }
-                { this.isStep(2) &&
-                    <div>
-                        <LayerCapabilitiesListing
-                            onSelect={(item) => mutator.setName(item.name)}
-                            capabilities={service.getCapabilities()} />
-                        <hr/>
-                        <Button onClick={() => this.setStep(1)}>Back</Button>
-                    </div>
-                }
-                { this.isStep(3) &&
-                    <div>
-                        <LayerDetails
-                            layer={layer} />
-                        <hr/>
-                        <Button onClick={() => this.setStep(2)}>Back</Button>
-                    </div>
-                }
-            </StyledRootEl>
-        );
+    if (typeof layer.layerName === 'undefined') {
+        return 2;
     }
+    return 3;
 }
+
+const LayerWizard = ({
+    mutator,
+    layer,
+    capabilities = [],
+    layerTypes = [],
+    loading,
+    children,
+    getMessage
+}) => {
+    let typeTitle = getMessage('wizard.type');
+    if (layer.type) {
+        typeTitle = `${typeTitle}: ${layer.type}`;
+    }
+    return (
+        <div>
+            { (layer.isNew || getStep(layer) !== 3) &&
+            <Steps current={getStep(layer)}>
+                <Step title={ typeTitle } />
+                <Step title={getMessage('wizard.service')} />
+                <Step title={getMessage('wizard.layers')} />
+                <Step title={getMessage('wizard.details')} />
+            </Steps>
+            }
+            { getStep(layer) === 0 &&
+                <LayerTypeSelection
+                    types={layerTypes || []}
+                    onSelect={(type) => mutator.setType(type)} />
+            }
+            { getStep(layer) === 1 &&
+                <div>
+                    <LayerURLForm
+                        layer={layer}
+                        loading={loading}
+                        service={mutator} />
+                    <hr/>
+                    <Button onClick={() => setStep(mutator, 0)}>{getMessage('cancel')}</Button>
```

Comment:
Create local reusable cancel button?

Reply:
Removed step specific back button and using a history(-1) type logic for it.